Minio/DataModel/Item.cs

```diff
@@ -54,4 +48,6 @@ public string ETag
             return dt;
         }
     }
+
+    [XmlElement] public List<MetadataItem> UserMetadata { get; set; }
```

Comment:
Yes, that is the goal. The scheme is supposed to have it there.
Maybe you meant "don't include it if empty" or maybe something else.
Please clarify.

Reply:
Why do you want explicitly annotate the property with an XML serialization attribute, but you don't do that for the other attributes. Either annotate all or none. Also it's a strange choice to use a `List` here, because metadata is key/value data, so a Dictionairy should be much more logical.

Reply:
The data structure `List<MetaDataItem>` is chosen this way, by the author of the original PR, I think that is because Server side sends the data the same way, and I did not change it.

The existing lint check and compiler enforces a collection definition in a class to be an interface. However, this causes exception when serializing for XML as interfaces are not serializable. So, having `[XElement]` was the only way I can have a serializable List.
I also addressed your question about if `UserMetadata` is empty. Now, if it is empty, it will be skipped and it is not going to be part of the XML.

I'd like to push this PR, and file any other requests/comments as a separate issue, if it is OK.

Reply:
Using a dictionary would be more logical, because:
1. You can use the index operator `[]` to fetch a specific key.
2. No key can every be specified more than once.

Reply:
I understand, and I agree with you. It would be make it,  less complicated. However, all other MinIO SDKs (golang, python, java et.al.) support listing objects with user/custom metadata when desired. This feature is not available in AWS, but (I searched a little) there were requests from customers for this feature to be added to MinIO.
We can discuss this in the meeting.

Reply:
I have no objection to provide the metadata while listing. I have two issues with the implementation:
- Userdata should be returned as a dictionary and not a list. Keys can only be used once, so a dictionary would make more sense.
- The implementation should be done the "MinIO" way using the `metadata=true` query parameter and not using the `StatObject` request that needs to be invoked for every object that is returned and kills performance.

We can discuss both in a meeting, but I don't see why. It's obvious for me to make this change. But maybe I'm missing something...

Reply:
I removed the `StatObject` api and started using `ListObjectsAsync` api command's `metadata=true` query parameter.